### ISSUE TITLE
Not Running Correctly

### ISSUE THREAD

**Issue by riotsmith** [dmarx/video-killed-the-radio-star#137]
Can anyone help me with the below. Can't get it to run and allow me to provide the API key.

---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
[<ipython-input-3-d10ed4e5584f>](https://localhost:8080/#) in <cell line: 24>()
     22 import time
     23 
---> 24 from omegaconf import OmegaConf
     25 
     26 

ModuleNotFoundError: No module named 'omegaconf'

---------------------------------------------------------------------------
NOTE: If your import is failing due to a missing package, you can
manually install dependencies using either !pip or !apt.

To view examples of installing some common dependencies, click the
"Open Examples" button below.

**Comment by jinnyhhh** [comment]
I just add 
`!pip install omegaconf`
at line 23, I think that might fix ur problem cause I just fixed mine by adding. Also make sure that at the end of this coding section, there are two lines about ' `from huggingface_hub import notebook_login'` and `'notebook_login()'` should be you need to let these two lines start from the head( no blank, just delete the blank in front of them). I think this problem is new cause its still working on Thursday, but yesterday I got thins problem too. This might help you, but this method isn't really professional actually, so if I have anything wrong, tell me plz and Im appreciate.

**Comment by riotsmith** [comment]
Thanks for your help so far! I'm still getting an error saying:
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
[<ipython-input-11-5792cf4df54c>](https://localhost:8080/#) in <cell line: 74>()
     72         pass
     73 
---> 74 from huggingface_hub import notebook_login
     75 
     76     # to do: if gdrive mounted, check for API token... somewhere on drive?

ModuleNotFoundError: No module named 'huggingface_hub'

I tried starting the code at the head for both lines and it didn't resolve.

Any ideas? Really appreciate your help

**Comment by riotsmith** [comment]
I managed to get round the previous problem but have since encountered another! Any help would be greatly appreciated!

---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
[<ipython-input-9-b60fe47f8a9c>](https://localhost:8080/#) in <cell line: 2>()
      1 import time
----> 2 from vktrs.utils import sanitize_folder_name
      3 from omegaconf import OmegaConf
      4 
      5 project_name = 'Peasants Vis' # @param {type:'string'}

ModuleNotFoundError: No module named 'vktrs'

---------------------------------------------------------------------------

**Comment by yyahav** [comment]
`vktrs` should be installed too (like omegaconf etc..) but it raises a new issue: #136

**Comment by danielamar101** [comment]
I am running into the same problem as well

**Comment by yyahav** [comment]
tbh the issue in #136 is not directly related to `vktrs`, it comes from failed building of `pytokenizations`:
[comment in #136](https://github.com/dmarx/video-killed-the-radio-star/issues/136#issuecomment-1529509540)

**Comment by yyahav** [comment]
I've found a fix, will create PR in a few minutes.
For the time being - in the `Installations` cell add the following line:
`!apt-get install cargo`
Just before:
`!pip install vktrs[api,hf]`

It's now working:

![image](https://user-images.githubusercontent.com/88027106/235438969-45594bec-2c63-429c-b2c8-009c2450c6df.png)
![image](https://user-images.githubusercontent.com/88027106/235439032-c03cb636-7657-4a0b-b774-4dc688382e37.png)
![image](https://user-images.githubusercontent.com/88027106/235439259-3a28e351-2e39-4c2d-b0fe-b69121f5fb42.png)

**Comment by yyahav** [comment]
This issue can be closed once #138 is merged.

**Comment by danielamar101** [comment]
This fixed it for me. I'm curious as to why, if you could be so kind as to explain.. I'm a bit new to python envs in general.

**Comment by yyahav** [comment]
@danielamar101 sure - when colab tries to `pip install vktrs` it has some dependencies already configured within the package like `omegaconf` and `pytokenizations`.

`pytokenizations` doesn't have a wheel for Colab (which is a Python package built for a specific platform), so when `pip` tries to install `pytokenizations` it downloads the source instead and tries to build it locally.

`pytokenizations` was written using Rust language, so the Rust builder should be used to build the package - hence the `Cargo`, which is Rust builder and package manager.
Because Cargo is not installed by default on Colab VM (Virtual Machine that runs your code), trying to build `pytokenizations` fails, so the whole operation of installing `vktrs` package fails, and that's why `omegaconf` is missing too (see the OP error in this issue).

Once Cargo is installed the build of pytokenizations is successful, and pip carries on to install the other dependencies and vktrs package.

**Comment by danielamar101** [comment]
Wow thank you for that explanation, will go a long way!
